File: src/Web/ClientApp/src/backendApi/clients/ThingsApiClient.ts
import { BackendRequestSender } from "src/backendApi/senders/BackendRequestSender";
import { IChangePropertyStateRequest } from "src/backendApi/models/request/things/IChangePropertyStateRequest";
import { IOperationResult } from "src/backendApi/models/response/OperationResult/IOperationResult";
import {IGetThingStateRequest} from "src/backendApi/models/request/things/IGetThingStateRequest";
import {IThingStateApiModel} from "src/backendApi/models/response/things/IThingStateApiModel";

export class ThingsApiClient {
    private static readonly apiUrlBase: string = "/api/ThingsApi/";
    private readonly backendRequestSender: BackendRequestSender = new BackendRequestSender();

    public changePropertyState(request: IChangePropertyStateRequest): Promise<IOperationResult<void>> {
        return this.backendRequestSender.sendPostRequest(`${ThingsApiClient.apiUrlBase}ChangePropertyState`, request);
    }

    public getThingState(request: IGetThingStateRequest): Promise<IOperationResult<IThingStateApiModel>> {
        return this.backendRequestSender.sendPostRequest(`${ThingsApiClient.apiUrlBase}GetThingState`, request);
    }
}

File: tests/Web.IntegrationTest/Utils/ApiClients/ThingsApiClient.cs
using System.Net.Http;
using System.Net.Http.Json;
using System.Threading.Tasks;
using Web.IntegrationTest.Utils.Parsers;
using Web.Models.OperationResults;
using Web.Models.Things;
using Web.Models.Things.Request;
using Web.Models.Things.Response;

namespace Web.IntegrationTest.Utils.ApiClients
{
    internal class ThingsApiClient : ApiClientBase
    {
        public ThingsApiClient(HttpClient httpClient, HttpResponseMessageParser httpResponseMessageParser)
            : base(httpClient, httpResponseMessageParser)
        {
        }

        protected override string ApiBaseUrl => "api/ThingsApi/";

        public async Task<OperationResult> ChangePropertyStateAsync(ChangePropertyStateRequest request)
        {
            HttpResponseMessage response = await HttpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, ApiBaseUrl + "ChangePropertyState")
            {
                Content = JsonContent.Create(request)
            });
            return await HttpResponseMessageParser.ParseResponseAsync<OperationResult>(response);
        }

        public async Task<OperationResult<ThingStateApiModel>> GetThingStateAsync(GetThingStateRequest request)
        {
            HttpResponseMessage response = await HttpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, ApiBaseUrl + "GetThingState")
            {
                Content = JsonContent.Create(request)
            });
            return await HttpResponseMessageParser.ParseResponseAsync<OperationResult<ThingStateApiModel>>(response);
        }

        public async Task<OperationResult<GetWorkspaceWithThingsResponse>> GetWorkspaceWithThingsAsync(GetWorkspaceWithThingsRequest requestModel)
        {
            HttpResponseMessage response = await HttpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, $"{ApiBaseUrl}GetWorkspaceWithThings")
            {
                Content = JsonContent.Create(requestModel)
            });
            return await HttpResponseMessageParser.ParseResponseAsync<OperationResult<GetWorkspaceWithThingsResponse>>(response);
        }
    }
}